keps/sig-instrumentation/1668-log-tracking/README.md

```diff
@@ -0,0 +1,366 @@
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Summary of Cases](#summary-of-cases)
+  - [Logging metadata](#logging-metadata)
+- [Design Details](#design-details)
+  - [Prerequisite](#prerequisite)
+  - [Design of ID propagation in apiserver](#design-of-id-propagation-in-apiserver)
+  - [Design of ID propagation (controller)](#design-of-id-propagation-controller)
+  - [Design of Mutating webhook(Out of tree)](#design-of-mutating-webhookout-of-tree)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha](#alpha)
+    - [Beta](#beta)
+    - [GA](#ga)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+This KEP proposes a method for adding new three unique logging meta-data into K8s component logs.
+It makes us more easy to identify specific logs related to an user request (such as `kubectl apply`) and object (such as Pod, Deployment).
+It is expected to reduce investigation cost greatly when trouble shoothing.
+
+### New three unique logging meta-data
+
+We use three meta-data. These meta-data have different features and are used for troubleshooting from different perspectives.
+
+| meta-data name | feature |
+| ------ | ------ |
+| trace-id | spans an user request. unique for user's request |
+| span-id | spans a controller action. unique for controller action |
+| initial-trace-id | spans the entire object lifecycle. unique for related objects |
```

Comment:
Thanks for your suggestion.
However, Our purpose of "initialtraceid" is to have the same ID across deployments and pods(and other k8s kind).
I think object uid cannot achieve it. What do you think?

Reply:
Let me give an example.  The current proposal is:

```
Deployment:
  name: "my-deployment"
  uid: 12345
  annotations:
    trace-annotation: 9999-8888-1 (trace id == 9999, span id == 8888, sampled == true)
    initial-trace-id: 9999

ReplicaSet:
  name: "my-rs"
  uid: 67823
  annotations:
    trace-annotation: 9999-7777-1 (trace id == 9999, span id == 7777, sampled == true)
    initial-trace-id: 9999

Pod:
  name: "my-pod"
  uid: 24680
  annotations:
    trace-annotation: 9999-6666-1 (trace id == 9999, span id == 6666, sampled == true)
    initial-trace-id: 9999
```

I'm suggesting:

```
Deployment:
  name: "my-deployment"
  uid: 12345
  annotations:
    trace-annotation: 9999-8888-1 (trace id == 9999, span id == 8888, sampled == true)
    initial-object-uid: 12345

ReplicaSet:
  name: "my-rs"
  uid: 67823
  annotations:
    trace-annotation: 9999-7777-1 (trace id == 9999, span id == 7777, sampled == true)
    initial-object-uid: 12345

Pod:
  name: "my-pod"
  uid: 24680
  annotations:
    trace-annotation: 9999-6666-1 (trace id == 9999, span id == 6666, sampled == true)
    initial-object-uid: 12345
```

But experimenting more, i'm not sure how useful uid would be, since you can't query for objects based on a uid...

Reply:
Thanks for the example, I can understand your suggestion.
Currently I have not been able to decide whether or not to use your suggestions, but if I find them useful, I will be happy to refer to them.